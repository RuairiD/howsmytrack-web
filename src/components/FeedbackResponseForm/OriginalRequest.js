import React from "react";

import { Col, Row, Typography } from "antd";
import { Div } from "lemon-reset";

import MediaEmbed from "../MediaEmbed/MediaEmbed";

const OriginalRequest = ({ mediaUrl, mediaType, feedbackPrompt }) => (
    <Div>
        <Row gutter={[16, 16]}>
            <Col>
                <MediaEmbed mediaUrl={mediaUrl} mediaType={mediaType} />
            </Col>
        </Row>
        {
            feedbackPrompt && (
                <Row gutter={[16, 16]}>
                    <Col>
                        <Typography.Text strong>The requester has said: </Typography.Text>
                        <Typography.Text>"{feedbackPrompt}"</Typography.Text>
                    </Col>
                </Row>
            )
        }
    </Div>
);

export default OriginalRequest;
