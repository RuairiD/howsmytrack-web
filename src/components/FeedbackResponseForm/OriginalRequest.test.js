import React from "react";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";
import OriginalRequest from "./OriginalRequest";

describe("OriginalRequest", () => {
    it("renders a MediaEmbed for the request's track", () => {
        const wrapper = shallow(
            <OriginalRequest
                mediaUrl="mediaUrl"
                mediaType="mediaType"
            />,
        );
        expect(toJson(wrapper)).toMatchSnapshot();
    });

    it("renders a feedback prompt, if one exists", () => {
        const wrapper = shallow(
            <OriginalRequest
                mediaUrl="mediaUrl"
                mediaType="mediaType"
                feedbackPrompt="feedbackPrompt"
            />,
        );
        expect(toJson(wrapper)).toMatchSnapshot();
    });
});
