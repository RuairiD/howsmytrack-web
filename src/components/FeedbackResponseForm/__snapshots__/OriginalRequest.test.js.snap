// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OriginalRequest renders a MediaEmbed for the request's track 1`] = `
<Div
  className=""
>
  <Row
    gutter={
      Array [
        16,
        16,
      ]
    }
  >
    <Col>
      <MediaEmbed
        mediaType="mediaType"
        mediaUrl="mediaUrl"
      />
    </Col>
  </Row>
</Div>
`;

exports[`OriginalRequest renders a feedback prompt, if one exists 1`] = `
<Div
  className=""
>
  <Row
    gutter={
      Array [
        16,
        16,
      ]
    }
  >
    <Col>
      <MediaEmbed
        mediaType="mediaType"
        mediaUrl="mediaUrl"
      />
    </Col>
  </Row>
  <Row
    gutter={
      Array [
        16,
        16,
      ]
    }
  >
    <Col>
      <Text
        strong={true}
      >
        The requester has said: 
      </Text>
      <Text>
        "
        feedbackPrompt
        "
      </Text>
    </Col>
  </Row>
</Div>
`;
